.github/workflows/ci.yml

```diff
@@ -89,77 +89,141 @@ jobs:
-  #       try-scenario:
-  #         - ember-3.28
-  #         - ember-4.0.0
-  #         - ember-4.4
-  #         - ember-release
-  #         - ember-beta
-  #         - ember-canary
-  #         - embroider-safe
-  #         - embroider-optimized
-
-  #   steps:
-  #     - uses: actions/checkout@v3
-  #     - uses: ./.github/actions/pnpm
-  #     - uses: ./.github/actions/download-built-package
-  #     - name: Run Tests
-  #       working-directory: ./testing/ember-app
-  #       run: >-
-  #         node_modules/.bin/ember try:one ${{ matrix.try-scenario }}
-  #         --skip-cleanup
-
-
-
-#   # https://github.com/changesets/action
-#   release:
-#     name: Release
-#     timeout-minutes: 5
-#     runs-on: ubuntu-latest
-#     if: github.ref == 'refs/heads/main'
-#     needs:
-#       - default_tests
-#       - floating_tests
-#       - typecheck
-#       - try_scenarios
-
-#     steps:
-#       - uses: actions/checkout@v3
-#         with:
-#           persist-credentials: false
-#       - uses: ./.github/actions/pnpm
-#       - uses: ./.github/actions/download-built-package
-#       - name: Create Release Pull Request or Publish to npm
-#         id: changesets
-#         uses: changesets/action@v1
-#         with:
-#           # This expects you to have a script called release which does a build for your packages and calls changeset publish
-#           publish: pnpm release
-#           title: "Release Preview"
-#         env:
-#           GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
-#           NPM_TOKEN: ${{ secrets.NPM_TOKEN }}
+  floating_tests:
+    name: Floating Deps Test
+    timeout-minutes: 5
+    runs-on: ubuntu-latest
+    needs: [build]
+    steps:
+      - uses: actions/checkout@v3
+      - uses: ./.github/actions/pnpm
+      - name: Install Dependencies (without lockfile)
+        run: rm pnpm-lock.yaml && pnpm install
+      - uses: ./.github/actions/download-built-package
+      - run: pnpm --filter test-app test:ember
+
+
+  try_scenarios:
+    name: ${{ matrix.try-scenario }}
+    timeout-minutes: 5
+    runs-on: ubuntu-latest
+    needs: [default_tests]
+
+    strategy:
+      fail-fast: false
+      matrix:
+        try-scenario:
+          - ember-4.8
+          - ember-release
+          - ember-beta
+          - ember-canary
+          - "'ember-release + embroider-optimized'"
+          - "'ember-lts-4.8 + embroider-optimized'"
+
+    steps:
+      - uses: actions/checkout@v3
+      - uses: ./.github/actions/pnpm
+      - uses: ./.github/actions/download-built-package
+      - name: Run Tests
+        working-directory: ./test-app
+        run: >-
+          node_modules/.bin/ember try:one ${{ matrix.try-scenario }}
+          --skip-cleanup
+
+
+
+  # https://github.com/changesets/action
+  release:
+    name: Release
+    timeout-minutes: 5
+    runs-on: ubuntu-latest
+    if: github.ref == 'refs/heads/main'
+    needs:
+      - default_tests
+      - floating_tests
+      - typecheck
+      - try_scenarios
```

Comment:
I guess we cannot split things up a bit into different workflows (like CI/release/docs) due to not being able to reference `needs` across workflows, right?

Reply:
correct